### PR TITLE
Add fingerprint to issues

### PR DESCRIPTION
**Pull request by gdiggs**
This adds fingerprints to issues that benefit from having the node name inside of the issue

@codeclimate/review

### CODE REVIEW

**Comment by pbrisbin** [comment]
LGTM, modulo CC issues.

File: JSONRenderer.php [hunk 1]
```diff
@@ -44,6 +47,14 @@ public function renderReport(Report $report)
                 );
             }
 
+            $name = $violation->getMethodName() ?: $violation->getFunctionName() ?: $violation->getClassName();
```
Comment (wfleming) [inline JSONRenderer.php:48]: 😨 lots of ternary stacking. We avoid this operator in other languages: I realize we don't really have a real style for PHP, but should we also avoid it here?

File: Fingerprint.php [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        $this->path = $path;
+        $this->rule = $rule;
+        $this->name = $name;
+    }
+
+    public function compute()
+    {
+        $fingerprint = NULL;
+
+
```
Comment (wfleming) [inline Fingerprint.php:26]: 💄 extra line

File: Fingerprint.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+class Fingerprint
+{
+    const OVERRIDE_RULES = [
+      "CyclomaticComplexity",
+      "Design/LongClass",
+      "Design/LongMethod",
+    ];
+    private $path;
+    private $rule;
```
Comment (wfleming) [inline Fingerprint.php:14]: Any reason these two are explicitly private but `$name` is not?

File: JSONRenderer.php (hunk 1 above)
Comment (gdiggs) [inline JSONRenderer.php:48]: Looks like there's a dedicated "Null Coalescing" operator, which is what I really want here. Will update

Comment (gdiggs) [inline JSONRenderer.php:48]: Scratch that - that's only available in PHP 7 which may not be available on alpine. Punting for now

**Comment by gdiggs** [comment]
@codeclimate/review cleaned up and ready for another look!

**Pull request merged**